python/ray/autoscaler/v2/instance_manager/config.py

```diff
@@ -18,129 +29,183 @@ class Provider(Enum):
 
 
 class IConfigReader(ABC):
-    """An interface for reading cluster config."""
+    """An interface for reading cluster config.
+
+    A utility class that converts the raw autoscaling configs into
+    various data structures that are used by the autoscaler.
+    """
 
     @abstractmethod
-    def get_cluster_config(self) -> ClusterConfig:
-        """Returns the cluster config."""
+    def get_autoscaling_config(self) -> "AutoscalingConfig":
+        """Returns the autoscaling config."""
         pass
 
 
-class NodeProviderConfig(object):
+class AutoscalingConfig(object):
```

Comment:
Renamed since this config maps directly to the autoscaling config (e.g. autoscaling speed, available instances types and etc), not only the node provider configs.